### PR TITLE
Document 3D navigation server in Godot 3.5

### PR DESCRIPTION
**Pull request by skyace65**
Documents the new 3D navigation server in Godot 3.5. It's a copy and paste of the 4.0 with some minor changes for 3.5. The old method is still documented, but it is stated that it will be removed in Godot 4.

### CODE REVIEW

File: tutorials/navigation/real_time_navigation_3d.rst [hunk 1]
```diff
@@ -6,46 +6,99 @@ Real Time Navigation (3D)
...
+-  :ref:`NavigationMeshInstance<class_NavigationMeshInstance>`
+-  :ref:`NavigationAgent<class_NavigationAgent>`
+-  :ref:`NavigationObstacle<class_NavigationObstacle>`
+
+The map and navigation regions
+------------------------------
+
+The "map" is the entire world for navigation, it's similar to "space" for
+the physics engine. It's comprised of navigation regions, these regions
+define what parts of the world that can be navigated around by navigation
```
Comment (mhilbrunner) [inline tutorials/navigation/real_time_navigation_3d.rst:23]:
```suggestion
define parts of the world that can be navigated around by navigation
```

File: tutorials/navigation/real_time_navigation_3d.rst [hunk 2]
```diff
@@ -6,46 +6,99 @@ Real Time Navigation (3D)
...
 
 Keep in mind that the navmesh shows where the center of an entity can
-go. For example, if you set the agent radius to 0.5 then your
+go. For example, if you set the agent radius to 0.5 then the
 navigation mesh will have a distance of 0.5 from any ledges or walls
 to prevent clipping into the wall or hanging off of the edge.
 
-Generating a path
+Navigation agents can moved from one region to another if they are next
+to each other. Additionally A baked navmesh can be moved at runtime and
```
Comment (mhilbrunner) [inline tutorials/navigation/real_time_navigation_3d.rst:55]:
```suggestion
to each other. Additionally a baked navmesh can be moved at runtime and
```

**Review by mhilbrunner — CHANGES_REQUESTED** [review]
Looks good, two minor things, besides that should be ready to merge :) Good work!

**Comment by skyace65** [comment]
Fixed.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks for helping filling out the 3.5 docs! 🎉 Good work :)

**Pull request merged**